src/NuGet.Core/NuGet.Frameworks/FrameworkReducer.cs

```diff
@@ -93,11 +94,38 @@ private NuGetFramework GetNearestInternal(NuGetFramework framework, IEnumerable<
                 // Remove lower versions of compatible frameworks
                 var reduced = ReduceUpwards(compatible);
 
-                // Reduce to the same framework name if possible
-                if (reduced.Count() > 1
-                    && reduced.Any(f => _fwNameComparer.Equals(f, framework)))
+                bool isNet6Era = StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, framework.Framework) && framework.Version.Major >= 6;
+
+                // Eliminate non-compatible xamarin frameworks
+                if (isNet6Era && framework.HasPlatform && reduced.Any(f => f.Framework.StartsWith("xamarin.", StringComparison.OrdinalIgnoreCase)))
                 {
-                    reduced = reduced.Where(f => _fwNameComparer.Equals(f, framework));
+                    reduced = reduced.Where(f =>
```

Comment:
I'm not sure this is sufficient. 

If I add a compatibility test, such as the below, it returns an incorrect result. 

I know it'll technically work for restore, because restore only ever calls the reducer but it creates a misleading API that could affect other features/other consumers of our APi.
```cs
        [Theory]
        [InlineData("net6.0-ios", "xamarin.mac", false)]
        public void Compatibility_FrameworksAreCompatible2(string project, string package, bool compatible)
        {
            // Arrange
            var projectFramework = NuGetFramework.Parse(project);
            var packageFramework = NuGetFramework.Parse(package);

            var compat = DefaultCompatibilityProvider.Instance;

            // Act & Assert
            Assert.Equal(compatible, compat.IsCompatible(projectFramework, packageFramework));
        }
```

Reply:
I guess I should move this logic into IsCompatible hmmm

Reply:
I've fixed IsCompatible and added tests :)